src/savegame_json.cpp

```diff
@@ -1447,6 +1447,15 @@ void vehicle::deserialize(JsonIn &jsin)
 
     data.read("parts", parts);
 
+    // we persist the pivot anchor so that if the rules for finding
+    // the pivot change, existing vehicles do not shift around.
+    // Loading vehicles that predate the pivot logic is a special
+    // case of this, they will load with an anchor of (0,0) which
+    // is what they're expecting.
+    data.read("pivot", pivot_anchor[0]);
+    pivot_anchor[1] = pivot_anchor[0];
+    pivot_rotation[1] = pivot_rotation[0] = fdir;
```

Comment:
Specifically, see pivot_displacement() and its callers. So that tiles don't shift around due merely to a change in pivot point, pivot_displacement works out where the old pivot point would be rotated to  when using the new pivot point, and the vehicle posx/posy is then shifted so that the old pivot point ends up at the same world coordinates.